myhoard/controller.py

```diff
@@ -1078,6 +1078,17 @@ def _purge_old_backups(self):
             owned_stream_ids = [sid for sid in self.state["owned_stream_ids"] if sid != backup["stream_id"]]
             self.state_manager.update_state(owned_stream_ids=owned_stream_ids)
 
+    def _get_oldest_binlog_time(self):
+        self._process_local_binlog_updates()
```

Comment:
This call seems out of place here. Is this actually needed for something?

Reply:
this updates the list of the binlogs so the function returns the current oldest binlog time

Reply:
Yes, I get that but the function name doesn't sound like it would be doing something like iterating through directory contents and parsing large files and I don't see why this call is needed here. We anyway periodically check the binlogs and it doesn't seem relevant whether the binlogs we have when running this code are few seconds old or not.

Reply:
We could rename the function. I don't really like the binlog_scanner to be honest I don't like side effects.

We do need this call, when we have a metric or a function call asking for information about something, we want the current truth not the truth as it was at some indeterminate  time in the past.

Reply:
OK yeah my bad I looked so hastily that didn't realize it was just purging binlogs right before sending the metric so refreshing the state is relevant. I'd be OK with just moving this call out of this function and doing it directly from the finally statement this function is getting called from.

Reply:
moved the call out of the function and calling it in the finally statement